src/shadow-embed.js

```diff
@@ -61,7 +62,10 @@ export function createShadowRoot(hostElement) {
   // Native support.
   const shadowDomSupported = getShadowDomSupportedVersion();
   if (shadowDomSupported == ShadowDomVersion.V1) {
-    return hostElement.attachShadow({mode: 'open'});
+    // NOTE(torch2424) need to have an id on the shadow root no matter what
+    const shadowRoot = hostElement.attachShadow({mode: 'open'});
```

Comment:
You mean `createShadowRootPolyfill` does? `hostElement.createShadowRoot()` doesn't. Generally, I think, this code simply doesn't belong here - it should probably go closer to where we instrument a shadow ampdoc.